app/connector/support/util/src/main/java/io/syndesis/connector/support/util/KeyStoreHelper.java

```diff
@@ -78,4 +91,24 @@ private static String generatePassword() {
         return new String(passwordChars, 0, passwordChars.length);
     }
 
+    public static KeyStore defaultKeyStore()
```

Comment:
To make this code portable and to take into account any [customizations configured](https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html#Customization) for the runtime.

I'd do something like:

```java
final KeyStore defaultKeystore = KeyStore.getInstance(KeyStore.getDefaultType());
defaultKeystore.load(null);

final TrustManagerFactory factory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
factory.init((KeyStore) null);

for (final TrustManager manager : factory.getTrustManagers()) {
    final X509TrustManager x509Manager = (X509TrustManager) manager;

    final X509Certificate[] issuers = x509Manager.getAcceptedIssuers();
    for (final X509Certificate issuer : issuers) {
        final String alias = issuer.getSerialNumber().toString();
        final TrustedCertificateEntry entry = new TrustedCertificateEntry(issuer);
        defaultKeystore.setEntry(alias, entry, null);
    }
}
```

and remove `CertificateUtil::defaultKeyStore` method.

Reply:
That's really cool and a lot more robust. Thx!